tests/backup_e2e.rs

```diff
@@ -0,0 +1,27 @@
+// This is an integration test file that lives in the tests/ directory
+// Integration tests verify that multiple components work together correctly
+mod common;  // Helper module for shared test utilities
+
+use common::{create_test_folder};  // Import test utilities
+use std::process::Command;  // For executing CLI commands
+
+#[test]
+fn test_backup_via_cli() -> Result<(), Box<dyn std::error::Error>> {
+    // Create temporary test directories using helper from common/mod.rs
+    let source_dir = create_test_folder();
+    let dest_dir = create_test_folder();
+    
+    // Test the CLI by actually running the binary with cargo run
```

Comment:
This is the only comment in the file that adds additional info or useful guidance, the rest should be deleted as they just repeat what is evident from the code.